Author: Zeeshan Ali
Origin: f9959f03c8241ea1ede5eb4079bec364301aad8d
Description: zb: Upgrade nix requirement to latest release
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/src/error.rs
+++ b/src/error.rs
@@ -106,9 +106,7 @@ impl From<io::Error> for Error {
 
 impl From<nix::Error> for Error {
     fn from(val: nix::Error) -> Self {
-        val.as_errno()
-            .map(|errno| io::Error::from_raw_os_error(errno as i32).into())
-            .unwrap_or_else(|| io::Error::new(io::ErrorKind::Other, val).into())
+        io::Error::from_raw_os_error(val as i32).into()
     }
 }
 
--- a/src/raw/socket.rs
+++ b/src/raw/socket.rs
@@ -85,8 +85,7 @@ impl Socket for UnixStream {
                 }
                 Ok((msg.bytes, fds))
             }
-            Err(nix::Error::Sys(e)) => Err(e.into()),
-            _ => Err(io::Error::new(io::ErrorKind::Other, "unhandled nix error")),
+            Err(e) => Err(e.into()),
         }
     }
 
@@ -104,8 +103,7 @@ impl Socket for UnixStream {
                 "failed to write to buffer",
             )),
             Ok(n) => Ok(n),
-            Err(nix::Error::Sys(e)) => Err(e.into()),
-            _ => Err(io::Error::new(io::ErrorKind::Other, "unhandled nix error")),
+            Err(e) => Err(e.into()),
         }
     }
 
--- a/src/utils.rs
+++ b/src/utils.rs
@@ -21,7 +21,7 @@ pub(crate) fn wait_on(fd: RawFd, flags:
     loop {
         match poll(&mut [pollfd], -1) {
             Ok(_) => break,
-            Err(nix::Error::Sys(e)) => {
+            Err(e) => {
                 if e == Errno::EAGAIN || e == Errno::EINTR {
                     // we got interupted, try polling again
                     continue;
